runtime/runtime/src/actions.rs

```diff
@@ -470,9 +478,52 @@ pub(crate) fn action_delete_account(
     // We use current amount as a pay out to beneficiary.
     let account_balance = account.as_ref().unwrap().amount();
     if account_balance > 0 {
-        result
-            .new_receipts
-            .push(Receipt::new_balance_refund(&delete_account.beneficiary_id, account_balance));
+        checked_feature!(
+            "protocol_feature_allow_create_account_on_delete",
+            AllowCreateAccountOnDelete,
+            current_protocol_version,
+            {
+                let sender_is_receiver = account_id == &delete_account.beneficiary_id;
+                let exec_gas = config.action_receipt_creation_config.send_fee(sender_is_receiver)
+                    + transfer_send_fee(
+                        &config.action_creation_config,
+                        sender_is_receiver,
+                        &delete_account.beneficiary_id,
+                        current_protocol_version,
+                    );
+                result.gas_burnt += exec_gas;
+                result.gas_used += exec_gas
+                    + config.action_receipt_creation_config.exec_fee()
+                    + transfer_exec_fee(
+                        &config.action_creation_config,
+                        &delete_account.beneficiary_id,
+                        current_protocol_version,
+                    );
+
+                result.new_receipts.push(Receipt {
+                    predecessor_id: account_id.clone(),
+                    receiver_id: delete_account.beneficiary_id.clone(),
+                    receipt_id: CryptoHash::default(),
```

Comment:
Receipt id must be globally unique.

Reply:
Discussed offline: receipt ids are assigned to all `new_receipts` in `apply_action_receipt` using `create_receipt_id_from_receipt` function.